liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -594,21 +594,32 @@ private void addArtifacts(org.eclipse.aether.graph.DependencyNode root, List<Fil
         List<Artifact> updatedArtifacts = new ArrayList<Artifact>();
         for (Dependency dep : dependencies) {
             boolean newDependency = true;
-            for (Dependency existingDep : existingDependencies) {
-                if (dep.getArtifactId().equals(existingDep.getArtifactId())) {
-                    newDependency = false;
-                    break;
+            try {
+                // resolve new artifact
+                Artifact artifact = getArtifact(dep.getGroupId(), dep.getArtifactId(), dep.getType(), dep.getVersion());
+
+                // match dependencies based on artifactId, groupId
+                for (Dependency existingDep : existingDependencies) {
+                    if (Objects.equals(artifact.getArtifactId(), existingDep.getArtifactId())
+                            && Objects.equals(artifact.getGroupId(), existingDep.getGroupId())) {
+                        if (Objects.equals(artifact.getVersion(), existingDep.getVersion())) {
+                            if (Objects.equals(artifact.getType(), existingDep.getType())) {
+                                newDependency = false;
+                                break;
+                            }
+                        }
+                    }
                 }
-            }
-            if (newDependency) {
-                log.debug("New dependency found: " + dep.getArtifactId());
-                try {
-                    Artifact artifact = getArtifact(dep.getGroupId(), dep.getArtifactId(), dep.getType(), dep.getVersion());
+                if (newDependency) {
+                    log.debug("New dependency found: " + artifact.toString());
                     updatedArtifacts.add(artifact);
-                } catch (MojoExecutionException e) {
-                    log.warn(e.getMessage());
                 }
+            } catch (MojoExecutionException e) {
+                log.warn(e.getMessage());
+            } catch (IllegalArgumentException e) {
+                log.warn("Illegal argument on " + dep.toString() + " " + e.getMessage());
```

Comment:
The following would be clearer to the user:
`log.warn("The dependency " + dep.toString() + " is not valid: " + e.getMessage());`

Reply:
If we do ```log.warn(dep.toString() + " is not valid: " + e.getMessage());``` the message will look like this: 
```
[WARNING] Dependency {groupId=, artifactId=test, version=1.0, type=jar} is not valid: groupId can neither be null, empty nor blank
```